src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -157,7 +157,12 @@ def acr_update_set(cmd,
     return client.update(resource_group_name, registry_name, parameters)
 
 
-def acr_login(cmd, registry_name, resource_group_name=None, username=None, password=None):  # pylint: disable=unused-argument
+def acr_login(cmd,
+              registry_name,
+              resource_group_name=None,  # pylint: disable=unused-argument
```

Comment:
Try removing resource_group_name from this function, seeing you don't use it. I think the function should still work.

Reply:
It's a breaking change, kept to deprecate soon, see https://github.com/Azure/azure-cli/pull/8185#pullrequestreview-190499729. The same for others.

Reply:
I see, I assumed that the parameter would still be visible. Ultimately, this module shouldn't add resource_group_name in _params.py seeing it is added to all commands as a global parameter [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/__init__.py).

Reply:
No it's not visible, but you still can provide it.

Reply:
Yes, I understand why we are leaving the parameter in, that's fine. I agree.

What I am saying is that the module OVERALL should not add resource_group as a local parameter in acr/_params.py because it is already a global parameter.  It would be good to make this change in the future, in a separate PR.

Reply:
These are legacy code written two years ago ...

Reply:
Besides, if I remove `resource_group_name`, the command complains when I specify it

```
>az acr login -n registry -g RG
az: error: unrecognized arguments: -g RG
```

Reply:
I already agreed with you @djyou. did you see my updated comment?

Reply:
> Yes, I understand why we are leaving the parameter in, that's fine. I agree.
> 
> What I am saying is that the module OVERALL should not add resource_group as a local parameter in [acr/_params.py](https://github.com/Azure/azure-cli/blob/c31e90fae295c5d717615eaf538f19562aef643a/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_params.py#L73) because it is already a global parameter [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/__init__.py#L249). It would be good to make this change in the future, in a separate PR.

^^^

Reply:
Yep, I get your point. Searching the repo shows that there are many command modules who register their own `resource_group_name` https://github.com/Azure/azure-cli/search?q=%22c.argument%28%27resource_group_name%27%2C+arg_type%3Dresource_group_name_type%29%22&unscoped_q=%22c.argument%28%27resource_group_name%27%2C+arg_type%3Dresource_group_name_type%29%22

I suspect the `resource_group_name` wasn't initially globally registered so many command modules registered their own `resource_group_name`, or they have non-default settings. But then extra effort is needed to clean them up. I don't see much value/harm regardless since it's just an extra line. /cc @tjprescott in case you have something to say here.

Reply:
Agreed. That is likely what happened.